src/platform/posix/tcp_ip_channel.c

```diff
@@ -52,32 +76,73 @@ static lf_ret_t TcpIpChannel_open_connection(NetworkChannel *untyped_self) {
   return LF_OK;
 }
 
-/**
- * @brief If is server: Try to accept
- * If is client: Try to connect
- */
-static lf_ret_t TcpIpChannel_try_connect(NetworkChannel *untyped_self) {
+static lf_ret_t TcpIpChannel_try_connect_server(NetworkChannel *untyped_self) {
   TcpIpChannel *self = (TcpIpChannel *)untyped_self;
 
-  /* Server -> Accept */
-  if (self->server) {
-    int new_socket;
-    struct sockaddr_in address;
-    socklen_t addrlen = sizeof(address);
+  int new_socket;
+  struct sockaddr_in address;
+  socklen_t addrlen = sizeof(address);
+
+  new_socket = accept(self->fd, (struct sockaddr *)&address, &addrlen);
+  if (new_socket >= 0) {
+    self->client = new_socket;
+    FD_SET(new_socket, &self->set);
+    return LF_OK;
+  } else {
+    if (errno == EWOULDBLOCK || errno == EAGAIN) {
+      LF_DEBUG(NET, "Accept failed. Try again. (errno=%d)", errno);
+      return LF_TRY_AGAIN;
+    } else {
+      LF_ERR(NET, "Accept failed. Unknown errno=%d", errno);
+      throw("Accept failed");
+      return LF_ERR;
+    }
+  }
+}
 
-    new_socket = accept(self->fd, (struct sockaddr *)&address, &addrlen);
-    if (new_socket >= 0) {
-      self->client = new_socket;
-      FD_SET(new_socket, &self->set);
+static lf_ret_t check_if_socket_is_writable(int fd) {
```

Comment:
This and other function should maybe start with `TcpIpChannel_`. At least this is our current code style as I can remember?
I personally prefer to have static functions start with an underscore, it makes it more clear, which functions are actually public for me and in the IDE, but better have it consistent through the project

Reply:
Hmm. The `TcpIpChannel_` prefix was initially meant for naming functions that had many implementations for different "subclasses". They are meant for implementing the corresponding functions in the public API (which is the function pointers on the struct). For purely static functions it might be different. But it might get messy if we dont pick a single convetion. I will follow your suggestion for now